stdlib/unittest/mock.pyi

```diff
@@ -74,21 +74,40 @@ class _Sentinel:
 
 sentinel: Any
 DEFAULT: Any
+_ArgsKwargs = tuple[tuple[Any, ...], Mapping[str, Any]]
+_NameArgsKwargs = tuple[str, tuple[Any, ...], Mapping[str, Any]]
+_ValueArgument = str | tuple[Any, ...] | Mapping[str, Any] | _ArgsKwargs | _NameArgsKwargs
+_CallValues = TypeVar("_CallValues", _ArgsKwargs, _NameArgsKwargs)
 
-class _Call(tuple[Any, ...]):
+class _Call(Generic[_CallValues]):
+    @overload
     def __new__(
         cls: type[Self],
-        value: Any = ...,
-        name: Any | None = ...,
+        value: _ValueArgument = ...,
+        name: str | None = ...,
         parent: Any | None = ...,
-        two: bool = ...,
+        two: Literal[True] = ...,
+        from_kall: bool = ...,
+    ) -> Self[_ArgsKwargs]: ...
```

Comment:
Mypy doesn't like this for some reason. Any ideas?

Reply:
TypeVars currently aren't subscriptable. We'd need [higher-kinded TypeVars](https://github.com/python/typing/issues/548) for this kind of syntax.

Reply:
Right, I forgot that `Self` is a TypeVar :) I'm thinking about just removing `_CallValues` to reduce complexity.

Reply:
Removed.